debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -394,7 +400,34 @@ protected void complete(SnapshotContext<MySqlOffsetContext> snapshotContext) {
      */
     @Override
     protected Optional<String> getSnapshotSelect(RelationalSnapshotContext<MySqlOffsetContext> snapshotContext, TableId tableId) {
-        return Optional.of(String.format("SELECT * FROM `%s`.`%s`", tableId.catalog(), tableId.table()));
+        String modifiedColumns = excludeColumns(tableId);
+        return Optional.of(String.format("SELECT %s FROM `%s`.`%s`", modifiedColumns, tableId.catalog(), tableId.table()));
+    }
+
+    @Override
+    protected String excludeColumns(TableId tableId) {
+        Table table = databaseSchema.tableFor(tableId);
+        List<String> columnNames = table.retrieveColumnNames().stream()
+                .filter(columnName -> connectorConfig.getColumnFilter().matches(tableId.catalog(), tableId.schema(), tableId.table(), columnName))
+                .collect(Collectors.toList());
+
+        if (columnNames.isEmpty()) {
+            LOGGER.info("Table " + tableId + " has no selected columns after applying column include/exclude lists");
```

Comment:
I'm not sure I agree with the logged message.  While the column include/exclude filters led to an empty column list in, we default to selecting the primary key columns as a baseline.  I would suggest perhaps that the message actually be something like:
```
All columns in table {} were excluded due to include/exclude lists, defaulting to selecting primary keys only.
```